File: test/fixtures/watchcallback/ios/Sources/WatchPlugin.swift
import Foundation
import Capacitor

@objc(WatchPlugin)
public class WatchPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "WatchPlugin"
    public let jsName = "Watch"
    public let pluginMethods: [CAPPluginMethod] = []

    @objc func getPosition(_ call: CAPPluginCall) {
        call.resolve(["lat": 1.0, "lng": 2.0])
    }

    @objc func watch(_ call: CAPPluginCall) {
        call.resolve(["id": "w1"])
    }

    @objc func stopWatch(_ call: CAPPluginCall) { call.resolve() }

    _objc func getSecurity(_ call: CAPPluginCall) {
        call.resolve(["state": "granted"])
    }
}

File: test/fixtures/watchcallback/android/src/main/java/com/example/watchcallback/WatchPlugin.java
package com.example.watchcallback;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "Watch")
public class WatchPlugin extends Plugin {
    @PluginMethod
    public void getPosition(PluginCall call) { call.resolve(new JSObject().put("lat", 1.0).put("lng", 2.0)); }
    @PluginMethod
    public void watch(PluginCall call) { call.resolve(new JSObject().put("id", "w1")); }
    @PluginMethod
    public void stopWatch(PluginCall call) { call.resolve(); }
    @PluginMethod
    public void getSecurity(PluginCall call) { call.resolve(new JSObject().put("state", "granted")); }
}
